### ISSUE TITLE
Further improve selection of best epitope in aggregate report

### ISSUE THREAD

**Issue by susannasiebert** [griffithlab/pVACtools#912]
From https://github.com/griffithlab/pVACtools/pull/870#issuecomment-1306756067 (credit @toddajohnson):

I was wondering about the impact of pVACtools's TSL filter last week, so I summarized genes in the Gencode build 38 (ENSEMBL 104) gtf for min. and max. TSL as well as if transcripts were labeled with other annotations used by Ensembl to prioritize gene models such as MANE, CCDS, etc.. Of 19670 protein-coding genes, 1043 only had transcripts with tslNA. From reviewing those tslNA only genes and Ensembl's TSL description, it looks to me like excluding tslNA would remove every single-exon gene. Of course, those may be small by size, but is that really the desired behavior? Also, by excluding tslNA and setting a default to TSL<=1 (while including tx. with no TSL annotation) would remove about 3000 (15%) of protein-coding genes. It seems that setting the ability to annotate using other Ensembl flags like MANE, APPRIS, Ensemble canonical, etc. might be more robust.

**Comment by malachig** [comment]
Yes, I think we should revisit how pVACseq chooses the "best" transcript:

From: https://github.com/wustl-oncology/analysis-wdls/issues/124

When there are multiple transcripts giving rise to equally good neoantigen candidates, pVACseq chooses a "best transcript". It seems that TSL status has a big impact on the choice. Essentially if there is a TSL=1 transcript, I believe that one will be chosen.

The logic for choosing the top transcript in Ensembl VEP is quite different from this. TSL is one of several considerations but other considerations take precedence over TSL. Consider the following example:

<img width="1275" alt="Image" src="https://github.com/user-attachments/assets/99cd6d58-a120-4ee3-83d0-531239bfdced" />

By choosing from TSL=1 transcripts we are often passing over the MANE select, canonical, refseq transcript. The top transcript according to Ensembl has a TSL>1 surprisingly often. I will open a new issue to start to expand on how we might rethink "best" transcript prioritization in pVAC.

Reviewing candidates with TSL>1 is currently one of the more frequent activities during manual genomic review of neoantigen candidates.

Some initial thoughts:

- If a `--flag-pick` transcript annotation is present that gives rise to best peptide, even if other transcripts also give the same peptide with better TSL, we could prioritize the flagged one.
- Other options might be to allow for custom prioritization based on MANE or Canonical status. 

We still want to prioritize based on what gives the best peptide.  But within a set of transcripts (pvac transcript set) that give rise to the same exact peptides, choosing the best from among these based on TSL is often sub-optimal.

**Comment by susannasiebert** [comment]
@toddajohnson We have now incorporated additional transcript filtering/prioritization criteria in pVACtools 6.0, which might be of interest to you. pVACseq and pVACsplice now take into account the MANE Select and Canonical status of a transcript for filtering, prioritizing, and tiering a neoantigen candidate. MANE Select, Canonical, and TSL status is evaluated according to the new `--transcript-prioritization-strategy` parameter. This parameter allows users to list one or more criteria (mane_select, canonical, tsl) to take into consideration. A transcript matching any of the specified criteria will pass our filtering.
